File: benchmark/speed_test.py
import time
import math

class Trade:
    def __init__(self, id, price, amount):
        self.id = id
        self.price = price
        self.amount = amount

def test_parsing():
    start = time.time()
    for i in range(100000):
        s = "{\"p\": 2500.50, \"v\": 100}"
        p = float(s[6:13])
    end = time.time()
    print(f"Tests A (Parsing): {(end-start)*1000:.2f} ms")

def test_math():
    start = time.time()
    result = 0
    for i in range(1000000):
        result += math.sqrt(i * 1.05) / 3.14159
    end = time.time()
    print(f"Tests B (Math): {(end-start)*1000:.2f} ms")

def test_memory():
    start = time.time()
    for i in range(500000):
        t = Trade("tx_0x123", 100.50, 1.0)
    end = time.time()
    print(f"Tests C (Memory): {(end-start)*1000:.2f} ms")

print("🐍 PYTHON MULTI-SCENARIO BENCHMARK")
test_parsing()
test_math()
test_memory()

File: benchmark/speed_test.rs
use std::time::Instant;

struct Trade {
    id: String,
    price: f64,
    amount: f64,
}

fn test_parsing() {
    let start = Instant::now();
    for _ in 0..100_000 {
        let s = "{\"p\": 2500.50, \"v\": 100}";
        let _p: f64 = s[6..13].parse().unwrap();
    }
    let duration = start.elapsed();
    println!("Tests A (Parsing): {:.2} ms", duration.as_secs_f64() * 1000.0);
}

fn test_math() {
    let start = Instant::now();
    let mut result = 0.0;
    for i in 0..1_000_000 {
        result += (i as f64 * 1.05).sqrt() / 3.14159;
    }
    let duration = start.elapsed();
    println!("Tests B (Math): {:.2} ms", duration.as_secs_f64() * 1000.0);
}

fn test_memory() {
    let start = Instant::now();
    for _ in 0..500_000 {
        let _t = Trade {
            id: String::from("tx_0x123"),
            price: 100.50,
            amount: 1.0,
        };
    }
    let duration = start.elapsed();
    println!("Tests C (Memory): {:.2} ms", duration.as_secs_f64() * 1000.0);
}

fn main() {
    println!("🦀 RUST MULTI-SCENARIO BENCHMARK");
    test_parsing();
    test_math();
    test_memory();
}
